api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -385,6 +291,10 @@ type AzureManagedControlPlaneStatus struct {
 	// OIDCIssuerProfile is the OIDC issuer profile of the Managed Cluster.
 	// +optional
 	OIDCIssuerProfile *OIDCIssuerProfileStatus `json:"oidcIssuerProfile,omitempty"`
+
+	// Version defines the Kubernetes version for the control plane instance.
+	// +optional
+	Version string `json:"version"`
```

Comment:
This field was added to fix an issue with updating the control plane via changing the version through the topology. This is because CAPI looks for the status.version field to be set to mark the amcp as ready 

Setting in CAPZ: https://github.com/willie-yao/cluster-api-provider-azure/blob/clusterclass-aks-final/controllers/azuremanagedcontrolplane_controller.go#L264
CAPI check: https://github.com/willie-yao/cluster-api/blob/fix-mp-cc-e2e/internal/contract/controlplane.go#L153

Reply:
Is this requirement documented in the CAPI provider contract?